clang/include/clang/Basic/Builtins.def

```diff
@@ -904,6 +904,32 @@ BUILTIN(__atomic_signal_fence, "vi", "n")
 BUILTIN(__atomic_always_lock_free, "bzvCD*", "nE")
 BUILTIN(__atomic_is_lock_free, "bzvCD*", "nE")
 
+// GNU atomic builtins with atomic scopes.
+ATOMIC_BUILTIN(__scoped_atomic_load, "v.", "t")
```

Comment:
I wonder if it would be better to preserve __atomic as the prefix, and move the _scope part to the name suffix

Reply:
Naming things is hard, we could do
```
__atomic_scoped_load
__scoped_atomic_load
__atomic_load_scoped
```
Unsure which is the best.